dashboard/test/models/section_test.rb

```diff
@@ -10,6 +10,49 @@ class SectionTest < ActiveSupport::TestCase
     @default_attrs = {user: @teacher, name: 'test-section'}
   end
 
+  test "sections are soft-deleted" do
+    assert_no_change("Section.with_deleted.count") do
+      @section.destroy
+      assert @section.reload.deleted?
+    end
+  end
+
+  test "destroying section destroys appropriate followers" do
+    already_deleted_follower = nil
+    Timecop.travel DateTime.new(2017, 2, 18, 12, 0, 0) do
+      already_deleted_follower = create :follower, section: @section
+      already_deleted_follower.destroy
+    end
+    follower = create :follower, section: @section
+
+    @section.reload.destroy
+
+    assert @section.reload.deleted?
+    assert follower.reload.deleted?
+    assert already_deleted_follower.reload.deleted?
+    assert_equal '2017-02-18 12:00:00 UTC', already_deleted_follower.deleted_at.to_s
+  end
+
+  test "restoring section restores appropriate followers" do
+    section = create :section
+    old_deleted_follower = create :follower, section: section
+    Timecop.travel Time.local(2012, 2, 18, 12, 0, 0) do
```

Comment:
Also, though it doesn't matter here, if you decorate your test with [freeze_time](https://github.com/code-dot-org/code-dot-org/blob/980e4629bc8829bd1cdc5c9f04a5b1a3010292db/dashboard/test/test_helper.rb#L204-L217), it will always run consistently at the same time of day.